src/overlays/actors/ovl_En_Skj/z_en_skj.c

```diff
@@ -39,201 +125,1497 @@ static ColliderCylinderInitType1 sCylinderInit = {
+        if (func_80106BC8(globalCtx) != 0) {
+            switch (globalCtx->msgCtx.choiceIndex) {
+                case 0: // Yes
+                    EnSkj_SetupTakeMask(this, globalCtx);
+                    break;
+                case 1: // No
+                    func_8010B720(globalCtx, 0x101D);
+                    EnSkj_SetupWaitForMaskTextClear(this);
+                    break;
+            }
+        }
+    }
+}
+
+void EnSkj_SetupTakeMask(EnSkj* this, GlobalContext* globalCtx) {
+    func_8010B720(globalCtx, 0x101E);
+    EnSkj_SetupAction(this, SKJ_ACTION_SARIA_SONG_TAKE_MASK);
+}
+
+void EnSkj_TakeMask(EnSkj* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 6) && (func_80106BC8(globalCtx) != 0)) {
+        Rupees_ChangeBy(10);
+        gSaveContext.itemGetInf[3] |= 0x200;
+        EnSkj_SetNaviId(this);
+        Player_UnsetMask(globalCtx);
+        Item_Give(globalCtx, ITEM_SOLD_OUT);
+        func_8010B720(globalCtx, 0x101F);
+        EnSkj_SetupWaitForMaskTextClear(this);
+    }
+}
+
+void EnSkj_SetupWaitForMaskTextClear(EnSkj* this) {
+    EnSkj_SetupAction(this, SKJ_ACTION_SARIA_SONG_WAIT_MASK_TEXT);
+}
+
+void EnSkj_WaitForMaskTextClear(EnSkj* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 6) && (func_80106BC8(globalCtx) != 0)) {
+        func_8002DF54(globalCtx, &this->actor, 7);
+        this->backfilpFlag = 1;
+        EnSkj_Backflip(this);
+    }
+}
+
+void EnSkj_SetupWrongSong(EnSkj* this) {
+    this->textId = 0x1041;
+    EnSkj_ChangeAnim(this, SKJ_ANIM_WAIT);
+    EnSkj_SetupAction(this, SKJ_ACTION_SARIA_SONG_WRONG_SONG);
+}
+
+void EnSkj_WrongSong(EnSkj* this, GlobalContext* globalCtx) {
+    if (D_80B01EA0 != 0) {
+        EnSkj_SetupWaitForTextClear(this);
+    } else {
+        func_8002F2CC(&this->actor, globalCtx, EnSkj_GetItemXzRange(this));
+    }
+}
+
+void EnSkj_SetupWaitForTextClear(EnSkj* this) {
+    EnSkj_SetupAction(this, SKJ_ACTION_SARIA_SONG_WAIT_FOR_TEXT);
+}
+
+void EnSkj_SariasSongWaitForTextClear(EnSkj* this, GlobalContext* globalCtx) {
+    u8 state;
+    Player* player;
+
+    if (((state = func_8010BDBC(&globalCtx->msgCtx)), player = PLAYER, state == 6) && func_80106BC8(globalCtx)) {
+        EnSkj_SetupWaitInRange(this);
+        player->stateFlags2 |= 0x800000;
+        player->unk_6A8 = (Actor*)sSmallStumpSkullKid.skullkid;
+    }
+}
+
+void EnSkj_OcarinaGameSetupWaitForPlayer(EnSkj* this) {
+    this->actor.flags &= ~1;
+    EnSkj_ChangeAnim(this, SKJ_ANIM_WAIT);
+    EnSkj_SetupAction(this, SKJ_ACTION_OCARINA_GAME_WAIT_FOR_PLAYER);
+}
+
+void EnSkj_OcarinaGameWaitForPlayer(EnSkj* this, GlobalContext* globalCtx) {
+    if (this->playerInRange) {
+        this->actor.flags |= 1;
+        EnSkj_SetupAction(this, SKJ_ACTION_OCARINA_GAME_IDLE);
+    }
+}
+
+s32 EnSkj_IsLeavingGame(EnSkj* this) {
+    s32 paramDecr = this->actor.params - 1;
+
+    if (sOcarinaMinigameSkullKids[paramDecr].unk0 == 2) {
```

Comment:
temp needed?

Reply:
Yes.  Also using a pad didnt work as it gets optimized out.